### PR TITLE
[utopia-620] [Backend] Unit Tests for helpers, utils, config service

### PR DESCRIPTION
**Pull request by kushal-arora-fw**
# Description

This PR includes the following proposed change(s):

- Added Unit Tests for the following helper files:
  - `base-helper`
  - `date-helper`
  - `pdf-helper`
  - `utils`
- Added Unit Tests for `config.service`

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring / Documentation
- [ ] Version change

## How Has This Been Tested?

By writing test cases and running `npm run test`

## Development Dependency Working Agreement
- [x] My code DOES NOT include the importing of new dependencies into the DPIA ecosystem
- [ ] My code DOES include the importing of new dependencies into the DPIA ecosystem
**If new dependencies are being introduced to the DPIA ecosystem:**
- [ ] The functionality of the dependency drastically reduces code complexity and makes my changes more easily maintainable and readible 
- [ ] The dependency being introduced does not contain multiple layers of nested dependencies introducing maintainability complexity to the DPIA ecosystem

## Frontend Development Changes
- [x] N/A
- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to project documentation or diagrams that reflect my changes
- [ ] New and existing unit tests pass locally with my changes
- [ ] My code follows Airbnb React Style Guidelines

## API Development Changes
- [ ] N/A
- [ ] I have performed a self-review of my own code
- [ ] My code follows standards and practices outlined in the BC Government API Development Guidelines
- [x] New and existing unit tests pass locally with my changes
- [ ] My changes includes Swagger documentation updates that reflect the changes I am introducing

## Definition of Done

Definition of Done

### CODE REVIEW

**Review by binggaofw — APPROVED** [review]
looks good to me

File: src/backend/test/unit/common/helpers/date-helper.spec.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+/**
+ * @Description
+ * This file tests the contents of common/helpers/data-helper.ts
+ */
+describe('DateHelper', () => {
+  /**
+   * @method shortDate
+   *
+   * @description
+   * This test suite validates that the method returns expected formatted data
```
Comment (antsand) [inline src/backend/test/unit/common/helpers/date-helper.spec.ts:12]: Would it be better to elaborate what the test is testing in terms of input/output

input: give a valid Date object 
          e.g. new Date('Wed Dec ....')
output: the date returned must be of the formant mm/dd/yy

What if we give a different input- null, random string, plain numbers, or a date in the short form but different formatting dd/mm/yyyy. What outputs should we expect?

If a new developer needs to expand this test case can we add to the description possible test cases to cover? 

Food for thought.

File: src/backend/test/unit/common/helpers/pdf-helper.spec.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * @Description
+ * This file tests the contents of common/helpers/pdf-helper.ts
+ */
+describe('PdfHelper', () => {
+  /**
+   * @method pugToPdfBuffer
+   *
+   * @description
+   * This test suite validates that the method does not throw error and returns the expected data
```
Comment (antsand) [inline src/backend/test/unit/common/helpers/pdf-helper.spec.ts:13]: Again, would add the general input that we are testing against and expected output

File: src/backend/test/unit/config/config.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    it('succeeds if all keys are present', () => {
+      const result = configServiceMock.ensureValues([
+        'test_key_1',
+        'test_key_2',
+      ]);
+
+      expect(result).toBe(configServiceMock);
+    });
+
+    // Scenario 2: Test fails if any ket is missing
```
Comment (antsand) [inline src/backend/test/unit/config/config.service.spec.ts:69]: spelling mistake - key*

File: src/backend/test/unit/config/config.service.spec.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    it('fails if any ket is missing', () => {
```
Comment (antsand) [inline src/backend/test/unit/config/config.service.spec.ts:70]: same here key*

**Review by antsand — APPROVED** [review]

**Review by Bottle7 — APPROVED** [review]

File: src/backend/test/unit/common/helpers/date-helper.spec.ts (hunk 1 above)
Comment (kushal-arora-fw) [reply src/backend/test/unit/common/helpers/date-helper.spec.ts]: > Would it be better to elaborate what the test is testing in terms of input/output

Cool, sure we can. I have added the input and outputs to the comments. <br/><br/>

> What if we give a different input- null, random string, plain numbers, or a date in the short form but different formatting dd/mm/yyyy. What outputs should we expect?

Updated the comment to reflect the null case to include defaults. random string, plain numbers, or a date in the short form won't be an accepted type. Typescript will take care of that for new devs.

And regarding the default date, we really can't test it as the current date is dynamic when the test runs. We may have to write similar logic to convert it to the date string, and the code to test code does not make sense and is an anti-pattern

**Pull request merged**